src/DssSpell.sol

```diff
@@ -106,226 +104,304 @@ contract DssSpellAction is DssAction {
+
+        // Add RWA015 contract to the changelog
+        DssExecLib.setChangelogAddress("RWA015",                     RWA015);
+        DssExecLib.setChangelogAddress("PIP_RWA015",                 pip);
+        DssExecLib.setChangelogAddress("MCD_JOIN_RWA015_A",          MCD_JOIN_RWA015_A);
+        DssExecLib.setChangelogAddress("RWA015_A_URN",               RWA015_A_URN);
+        DssExecLib.setChangelogAddress("RWA015_A_JAR",               RWA015_A_JAR);
+        DssExecLib.setChangelogAddress("RWA015_A_INPUT_CONDUIT_URN", RWA015_A_INPUT_CONDUIT_URN);
+        DssExecLib.setChangelogAddress("RWA015_A_INPUT_CONDUIT_JAR", RWA015_A_INPUT_CONDUIT_JAR);
+        DssExecLib.setChangelogAddress("RWA015_A_OUTPUT_CONDUIT",    RWA015_A_OUTPUT_CONDUIT);
+
+        // Add RWA015 to ILK REGISTRY
         IlkRegistryAbstract(REGISTRY).put(
             ilk,
-            MCD_JOIN_RWA014_A,
-            RWA014,
-            GemAbstract(RWA014).decimals(),
-            RWA014_REG_CLASS_RWA,
+            MCD_JOIN_RWA015_A,
+            RWA015,
+            GemAbstract(RWA015).decimals(),
+            RWA015_A_REG_CLASS_RWA,
             pip,
             address(0),
-            "RWA014-A: Coinbase Custody",
-            GemAbstract(RWA014).symbol()
+            "RWA015-A: BlockTower Andromeda",
+            GemAbstract(RWA015).symbol()
         );
-    }
 
-    function actions() public override {
+        // ----- Additional ESM authorization -----
+        DssExecLib.authorize(MCD_JOIN_RWA015_A,          MCD_ESM);
+        DssExecLib.authorize(RWA015_A_URN,               MCD_ESM);
+        DssExecLib.authorize(RWA015_A_OUTPUT_CONDUIT,    MCD_ESM);
+        DssExecLib.authorize(RWA015_A_INPUT_CONDUIT_URN, MCD_ESM);
+        DssExecLib.authorize(RWA015_A_INPUT_CONDUIT_JAR, MCD_ESM);
+
+        // --- Bootstrap ---
+        RwaUrnLike(RWA015_A_URN).hope(address(this));
+        RwaOutputConduitLike(RWA015_A_OUTPUT_CONDUIT).hope(address(this));
+        RwaOutputConduitLike(RWA015_A_OUTPUT_CONDUIT).mate(address(this));
+        RwaInputConduitLike(RWA015_A_INPUT_CONDUIT_URN).mate(address(this));
+        RwaInputConduitLike(RWA015_A_INPUT_CONDUIT_JAR).mate(address(this));
+
+        // Lock RWA015 Token in the URN
+        GemAbstract(RWA015).approve(RWA015_A_URN, 1 * WAD);
+        RwaUrnLike(RWA015_A_URN).lock(1 * WAD);
+        // Draw until the current debt ceiling
+        RwaUrnLike(RWA015_A_URN).draw(RWA015_A_LINE * WAD);
+
+        // Pick the destination for the assets
+        RwaOutputConduitLike(RWA015_A_OUTPUT_CONDUIT).pick(RWA015_A_CUSTODY);
+        // Swap Dai for the chosen stablecoin through the PSM and send it to the picked address.
+        // For Goerli we push only 100 Dai
+        RwaOutputConduitLike(RWA015_A_OUTPUT_CONDUIT).push(100 * WAD);
+        // For Mainnet we push the entire balance
+        // RwaOutputConduitLike(RWA015_A_OUTPUT_CONDUIT).push();
```

Comment:
Please remember to use this 2.5M value on mainnet

Reply:
For mainnet we will use the commented-out line below: `push()` (no parameters = push all outstanding balance).